smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/events/MessageUnmarshallableGenerator.kt

```diff
@@ -0,0 +1,210 @@
+                writer.openBlock(
+                    "static var unmarshal: \$N<\$N> {", "}",
+                    ClientRuntimeTypes.EventStream.UnmarshalClosure,
+                    streamSymbol,
+                ) {
+                    writer.openBlock("{ message in", "}") {
+                        writer.write("switch try message.type() {")
+                        writer.write("case .event(let params):")
+                        writer.indent {
+                            writer.write("switch params.eventType {")
+                            streamShape.eventStreamEvents(ctx.model).forEach { member ->
+                                writer.write("case \"${member.memberName}\":")
+                                writer.indent {
+                                    renderDeserializeEventVariant(ctx, streamSymbol, member, writer)
+                                }
+                            }
+                            writer.write("default:")
+                            writer.indent {
+                                writer.write("return .sdkUnknown(\"error processing event stream, unrecognized event: \\(params.eventType)\")")
+                            }
+                            writer.write("}")
+                        }
+                        writer.write("case .exception(let params):")
+                        writer.indent {
+                            writer.write(
+                                "let makeError: (\$N, \$N) throws -> \$N = { message, params in",
+                                ClientRuntimeTypes.EventStream.Message,
+                                ClientRuntimeTypes.EventStream.ExceptionParams,
+                                SwiftTypes.Error
+                            )
+                            writer.indent {
+                                writer.write("switch params.exceptionType {")
+                                streamShape.eventStreamErrors(ctx.model).forEach { member ->
+                                    writer.write("case \$S:", member.memberName)
+                                    writer.indent {
+                                        renderReadToValue(writer, member)
+                                        writer.write("return value")
+                                    }
+                                }
+                                writer.write("default:")
+                                writer.indent {
+                                    writer.write("let httpResponse = HttpResponse(body: .data(message.payload), statusCode: .ok)")
+                                    writer.write(
+                                        "return \$L(httpResponse: httpResponse, message: \"error processing event stream, unrecognized ':exceptionType': \\(params.exceptionType); contentType: \\(params.contentType ?? \"nil\")\", requestID: nil, typeName: nil)",
+                                        customizations.unknownServiceErrorSymbol,
```

Comment:
The `unknownServiceErrorSymbol` provided by customizations replaces the hard-coded reference to an AWS type so that this file can be moved to smithy-swift.

(Same change in one other place in this file)